subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/ArtifactTransform.java

```diff
@@ -28,14 +28,6 @@
  * <li>The objects provided to {@link org.gradle.api.ActionConfiguration#setParams(Object...)}.</li>
  * </ul>
  *
- * <p>A property annotated with {@link TransformParameters} will receive the object provided by {@link ArtifactTransformSpec#getParameters()}.
- *
- * <p>A property annotated with {@link PrimaryInput} will receive the <em>primary input</em> location, which is the file or directory that the transform should be applied to.
- *
- * <p>A property annotated with {@link Workspace} will receive the <em>workspace</em> location, which is the directory that the transform should write its output files to.
- *
- * <p>A property annotated with {@link PrimaryInputDependencies} will receive the <em>dependencies</em> of its primary input.
- *
  * @since 3.4
  */
 public abstract class ArtifactTransform {
```

Comment:
Is it time already to deprecate this guy?

Reply:
I wondered about that as well. Do we need to de-incubate the new API in order to deprecate the old one?

I'll create a card for the deprecation.